modin/config/envvars.py

```diff
@@ -415,6 +415,20 @@ class MinPartitionSize(EnvironmentVariable, type=int):
     varname = "MODIN_MIN_PARTITION_SIZE"
     default = 32
 
+    @classmethod
+    def put(cls, value):
+        """
+        Set ``MinPartitionSize`` with extra checks.
+
+        Parameters
+        ----------
+        value : int
+            Config value to set.
+        """
+        if value <= 0:
+            raise ValueError(f"Min partition size should be > 0, passed value {value}")
```

Comment:
@YarShev We have check here for the case you talking about.

Reply:
I am talking about other places where we call `MinPartitionSize.get()`. If a user sets a value via the env var (not via `MinPartitionSize.put(value)`, the check you specified in `MinPartitionSize.put()` won't be called.

Reply:
@YarShev got it, I was relying on the idea that the only intended way to set values is `put` function, however this can happens in `get` function under different conditions.

@vnlitvinov it seems to me that it's better to rewrite these classes so that the setting of values goes only through `put` function. What do you think about?

Reply:
This seems like a different problem. You're talking about verification of the correctness of the input.

There is already a mechanism which is specified per-type only that seems to be doing part of what you want, but for incoming values (like strings which we then cast to ints).
Seems like we should add similar mechanism to the values right before setting the internal value.

I propose to do it in a separate PR for all the values which should have such restrictions (like CPU count being >=1, etc.)

Reply:
Yes, adding such a mechanism looks as a separate issue.

Reply:
For now, I would suggest adding the assert to `get` and remove it if necessary when that mechanism is added.

Reply:
`assert` is added for `get` method. Ready for review.

Reply:
@anmyachev, please create an issue for adding that mechanism.